### DISCUSSION TITLE
Quarto Shiny Dynamic Container - how to publish developed document

### DISCUSSION THREAD

**Discussion by gmouawad** [quarto-dev/quarto-cli#4457] [Q&A]
Hello

We have installed on a container the quarto dynamic container.
if I have a quarto document developed, how I can publish it on that container?

if there is another way to publish critical documents/reports internally (on premises) please advice

Thanks in advance

**Comment by mcanouil** [comment]
It's not very clear to me what's your setup ...
What is "container", a Docker container? Something else?
And what's a "quarto dynamic container"?
What do you mean by "publish"? Make them available in a web service of sort?
What type of "documents/reports"? HTML? PDF? Word? PowerPoint?

If "documents/reports" are HTML and if by "publish" you mean make the rendered documents available on a web server which is a "container", then simply `scp`/`cp`/`sftp`/etc. them to where your web server is (no matter what is it).

If you want a more accurate/detailed answer, please elaborate on your needs and your setup.

**Reply by gmouawad** [reply]
Hello @mcanouil ,

yes docker container
dynamic quarto: https://hosting.analythium.io/how-to-set-up-quarto-with-docker-part-2-dynamic-content/
yes publish the document as a webpage on premises
of course HTML. pdf output can be rendered locally

So in short terms, I am developing a quarto document HTML, that reads data using python and deliver analysis information.
I need to have this page to be available for a group of people to access it and refresh it with live data.

This is my needs, I am looking on how to accomplish that.

Thanks and sorry for the lack of info provided in my initial description

**Reply by mcanouil** [reply]
If you are using shiny (R) then you need to have a shiny server running.
Note that a shiny server is also a web server which can host static HTML or shiny documents (Quarto or RMarkdown, or regular shiny app).

What you have to do is:
1. setup and run a container which has: R, shiny, Quarto, and all tools/packages needed.
2. Then either:
   - mount a shared volume to the container where you can add your documents.
   - SCP/SFTP your documents to inside the container.

But all this is quite independent of Quarto CLI.
It is mostly IT stuff regarding Docker and how to host Shiny app/documents.

**Reply by gmouawad** [reply]
ok @mcanouil thanks a lot. So nothing specific to quarto. Appreciate your help so much.

Thanks

**Reply by mcanouil** [reply]
Indeed (beside Quarto install).
